### PR TITLE
Refactor member redux incuding creating a MemberPanel component for e…

### PR DESCRIPTION
**Pull request by esteban737**
…asy component updating, altering member redux actions to include common functionality, and move firebase calls to member services

Fixes #

## Description

Simplifying member redux by removing unnecessary visitedMember state and saving members in commonly used formats such as sorted alphabetically and sorted by rank. Created MemberPanel component for modular use between different components. Moved firebase calls in members redux to members services.

## Types of changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other  Refactoring

## Checklist:

- [x] My code follows the code style of this project found on [Contribution guidelines](https://github.com/SHPEUCF/shpeucfapp/wiki/Contribution-guidelines).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by idelmr — CHANGES_REQUESTED** [review]
Didn't comment on any of the refactoring needed for the member files or any code that was like that before that can improved, e.g.:
* destructuring
* spacing
* unnecessary `return`s
* unnecessary multiline code

File: src/services/members.js [hunk 1]
```diff
@@ -1,72 +1,99 @@
...
-				let updates = snapshot.val();
+export const getAllMemberAccountsandRankings = listener => new Promise(resolve =>
+	firebase.database().ref('/users/')[listener ? 'on' : 'once']('value', snapshot => {
+		const allMemberAccounts = {};
+		let rankedIds = [];
 
-				members.forEach(memberId => {
-					updates[memberId][privilegeChanged] = value;
-				});
+		_.orderBy(snapshot.val(), ['lastName', 'firstName'], ['asc', 'asc']).forEach(member => allMemberAccounts[member.id] = member);
```
[the remark this answers is not in the document]
Comment (esteban737) [reply src/services/members.js:78]: They actually do keep they order somehow if you test it. Basically originally I had a sortedIDs array like rankedIDs but I thought about it and I dont think we'd ever really need the original ordering

[the remark this answers is not in the document]
Comment (idelmr) [reply src/services/members.js:78]: Got it 👍 Although we could probably reduce the code to this. We could remove the `reduce` if it's too much.
```js
const allMemberAccounts = _.orderBy(snapshot.val(), ['lastName', 'firstName'], ['asc', 'asc'])
	.reduce((allMembers, member)) => ({ ...allMembers, [member.id]: member }), {});
let rankedIds = _.orderBy(allMemberAccounts, ['points'], ['desc']).map(({ id }) => id);

for (let pastPoints = 0, pastRank = 1, i = 0; i < rankedIds.length; i++) {
	const { rank, points } = allMemberAccounts[rankedIds[i]];

	pastRank = (points === pastPoints) ? pastRank : (points !== 0) ? i + 1 : rankedIds.length;
	pastPoints = points;
}

resolve({ allMemberAccounts, rankedIds });
```

[the remark this answers is not in the document]
Comment (esteban737) [reply src/services/members.js:78]: If you're gonna declare variables in one line in the for loop, you might as well let us declare them in one line anywhere

**Review by idelmr — APPROVED** [review]

**Pull request merged**